### PR TITLE
Split zfs related grain off into a seperate module

### PR DESCRIPTION
**Pull request by sjorge**
### What does this PR do?
This PR will remove the zfs/zpool grains from ```salt.grains.core```, additionally it introduces a new ```zfs_support``` grain that is now used by salt.modules.zfs and salt.modules.zpool so they can skip the ugly do we need to load bits.

### What issues does this PR fix or reference?
n/a

### Previous Behavior
zfs/zpool grains were in the main salt.grains.core which was hard to manage and getting confusing.

### New Behavior
zfs/zpool related grains now live in salt.grains.zfs which is much easier to manage and debug.

### Tests written?
No

### Commits signed with GPG?
No

### CODE REVIEW

**Comment by sjorge** [comment]
In the past I already split of the mdata stuff from salt.grains.core but it looks like I left most of the smartos specific grains still in there. I though I had already moved them too. I will do a seperate PR for those.

File: salt/modules/zfs.py [hunk 1]
```diff
@@ -51,42 +62,6 @@ def _check_features():
...
-    elif on_solaris:
-        # not using salt.utils.path.which('zfs') to keep compatible with others
-        cmd = 'which zfs'
-
-    if cmd and salt.modules.cmdmod.retcode(
-        cmd, output_loglevel='quiet', ignore_retcode=True
-    ) == 0:
-        return 'zfs'
-
-    if __grains__['kernel'] == 'OpenBSD':
```
Comment (sjorge) [inline salt/modules/zfs.py]: The special exception for OpenBSD is no longer needed as we only check zfs-fuse support on linux were it makes sense.

With the code in salt.grains.zfs I do not see the service warnings anymore was the issues why the extra backout for OpenBSD was added here.

**Review by cachedout — APPROVED** [review]

**Comment by cachedout** [comment]
@sjorge Some failed tests here, I'm afraid.

**Comment by sjorge** [comment]
Aah yep, looks like a test in bsd_memstat seems to want to provide mock _zpool_data, since it is no longer in salt.grains.core I removed the mock. Let's see if it passes now.

**Comment by sjorge** [comment]
Looks like the issue with the bsd_memstat test is now fixed, it only shows the ```Tornado IOLoop errors``` error it shows on all PR for me now. Which seems to be unrelated.

**Comment by rallytime** [comment]
@sjorge Ah, I think after merging your other PR, it caused conflicts here. Can you rebase this?

**Comment by sjorge** [comment]
I’ll try and rebase it tonight

**Comment by sjorge** [comment]
@rallytime done, I had to skip the first commit and then redo the same action afterwards. The diff before and after the rebase is the same. 

I'm not sure jenkins can deal with a force push to the zfs_rains_split branch though, so let's see how the CI run goes.

**Review by rallytime — APPROVED** [review]

**Pull request merged**